packages/react-components/src/atoms/TextField.tsx

```diff
@@ -214,41 +189,44 @@ const TextField: React.FC<TextFieldProps> = ({
         maxLength={maxLength}
         pattern={pattern}
         value={value}
-        onChange={({ currentTarget: { value: newValue } }) => {
-          setIsDirty(true);
-          return onChange(newValue);
-        }}
+        onChange={({ currentTarget: { value: newValue } }) =>
+          onChange(newValue)
+        }
         css={[
           styles,
           textFieldStyles,
           enabled || disabledStyles,
 
-          debouncedIndicateValid && validStyles(validGifUrl),
           validationMessage && invalidStyles,
           !labelIndicator && { gridColumn: '1 / span 2' },
-          customIndicator &&
-            customIndicatorPadding(
-              getSvgAspectRatio(customIndicator),
-              customIndicatorPosition,
-            ),
-          loading && loadingStyles,
+
+          leftIndicator &&
+            getIndicatorPadding(getSvgAspectRatio(leftIndicator), 'left'),
```

Comment:
I think this could be simplified further. Return the padding string and set it to padding left or right where appropriate